sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -119,4 +121,32 @@ object IntervalUtils {
       case _: IllegalArgumentException => null
     }
   }
+
+  /**
+   * Gets interval duration
+   *
+   * @param cal - the interval to get duration
+   * @param targetUnit - time units of the result
+   * @param daysPerMonth - the number of days per one month
+   * @return duration in the specified time units
+   */
+  def getDuration(
+      cal: CalendarInterval,
+      targetUnit: TimeUnit,
+      daysPerMonth: Int = 31): Long = {
```

Comment:
Then we can remove the default value? The caller side should explicitly specify that it wants 31 days per month and why.

Reply:
I suggested the default as that is the only value used now. I think callers that specify otherwise can explain why (later)

Reply:
then at least we need to explain why 31 is picked as the default value